pkg/repository/postgres/dbsqlc/tenant_invites.sql

```diff
@@ -0,0 +1,93 @@
+-- name: CreateTenantInvite :one
+INSERT INTO "TenantInviteLink" (
+    "id",
+    "tenantId",
+    "inviterEmail",
+    "inviteeEmail",
+    "expires",
+    "role"
+) VALUES (
+    gen_random_uuid(),
+    @tenantId::uuid,
+    @inviterEmail::text,
+    @inviteeEmail::text,
+    @expires::timestamp,
+    @role::"TenantMemberRole"
+) RETURNING *;
+
+-- name: CountActiveInvites :one
+SELECT
+    COUNT(*)
+FROM
+    "TenantInviteLink"
+WHERE
+    "tenantId" = @tenantId::uuid
+    AND "status" = 'PENDING'
+    AND "expires" > now();
+
+-- name: GetExistingInvite :one
+SELECT
+    *
+FROM
+    "TenantInviteLink"
+WHERE
+    "inviteeEmail" = @inviteeEmail::text
+    AND "tenantId" = @tenantId::uuid
+    AND "status" = 'PENDING'
+    AND "expires" > now();
+
+-- name: GetInviteById :one
+SELECT
+    *
+FROM
+    "TenantInviteLink"
+WHERE
+    "id" = @id::uuid;
+
+-- name: ListTenantInvitesByEmail :many
+SELECT
+    *
+FROM
+    "TenantInviteLink"
+WHERE
+    "inviteeEmail" = @inviteeEmail::text
+    AND "status" = 'PENDING'
+    AND "expires" > now();
+
+-- name: ListInvitesByTenantId :many
+SELECT
+    *
+FROM
+    "TenantInviteLink"
+WHERE
+    "tenantId" = @tenantId::uuid
+    AND (
+        sqlc.narg('status')::"InviteLinkStatus" IS NULL
+        OR "status" = sqlc.narg('status')::"InviteLinkStatus"
+    )
+    AND (
+        CASE WHEN sqlc.narg('expired')::boolean IS NULL THEN TRUE
+        -- otherwise, if expired is true, return only expired invites, and vice versa
+        ELSE sqlc.narg('expired')::boolean = ("expires" < now())
+        END
+    )
+    AND (
+        sqlc.narg('expired')::boolean IS NULL
+        OR "expires" < now()
+    );
```

Comment:
I'm not sure I'm following this logic - I'm reading the first clause here as "If the expired flag is set to true, then only match rows that have expired, otherwise if it's false match only non-expired rows, otherwise match all rows" but then the second clause is "match only expired rows if the expired flag is set, regardless of if it's true or false. otherwise match all rows"

what am I missing here?

Reply:
whoops, you're totally right -- it should only be the first clause:

```sql
AND (
        CASE WHEN sqlc.narg('expired')::boolean IS NULL THEN TRUE
        -- otherwise, if expired is true, return only expired invites, and vice versa
        ELSE sqlc.narg('expired')::boolean = ("expires" < now())
        END
    )
```

Just to confirm this makes sense -- if we don't set `expired`, this is a no-op. otherwise, if `expired=True` then we select for rows whose `expires` is before now. if `expired=False` we select for rows whose expires is gte now.

Reply:
yeah, that sounds right to me 👍